docs/DeveloperGuide.adoc

```diff
@@ -387,6 +387,40 @@ image::UndoRedoActivityDiagram.png[width="650"]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
+// tag::addpicture[]
+=== Add Picture feature
+==== Current Implementation
+
+The picture mechanism is facilitated by the new `PictureCommand`.
+It extends `Command` with an execution to set a picture, stored internally in `Person` as `picture`.
+
+Given below is an example usage scenario and how the picture mechanism behaves at each step.
+
+Step 1. The user launches the application and already has at least one client's contact in InsuRen.
+
+image::PictureCommand1StateDiagram.png[width="800"]
+
+Step 2. The user executes `pic 4 l//images/david.jpg` to add a picture for David. The `pic` command calls `Model#getFilteredPersonList()` to retrieve the list of contacts and filters `index` 4. It then calls `Picture#setPicture()` to update the picture for the contact. Finally, `Model#commitAddressBook()` is called, causing the modified state of the address book after the `pic 4 l//images/david.jpg` command executes to be saved.
+
+image::PictureCommand2StateDiagram.png[width="800"]
+
+The following activity diagram summarizes what happens when a user executes the `PictureCommand`:
+
+image::PictureCommandActivityDiagram.png[width="800"]
+
+==== Design Considerations
+
+===== Aspect: How picture is stored
+
+* **Alternative 1 (current choice):** `Person` has a picture field.
+** Pros: `Picture` can have it's own `Picture#isValidPicture()` method to validate the input. It is consistent with the other fields within `Person`.
+** Cons: More memory is used as there is a need to store an object. A new `Picture` class have to be made and implemented.
```

Comment:
Grammar nitpick: "A new `Picture` class has to be made and implemented" instead of "have"